t/sys/chdir.c

```diff
@@ -0,0 +1,361 @@
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* try getcwd with a NULL buffer but short size,
+     * should fail with ERANGE */
+    str = getcwd(NULL, len);
+    ok(str == NULL && errno == ERANGE, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    if (str != NULL) {
+        free(str);
+    }
+    errno = 0;
+
+    /* try getcwd with a buffer and 0 size, should fail with EINVAL */
+    str = getcwd(path, 0);
+    ok(str == NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    if (str != NULL) {
+        free(str);
+    }
+    errno = 0;
+
+    /* try getcwd with a NULL buffer and 0 size,
+     * getcwd should allocate buffer */
+    str = getcwd(NULL, 0);
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* change to unifyfs/dir1 */
+    rc = chdir(buf);
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, buf, strerror(errno));
+
+    /* check that we're in unifyfs/dir1 */
+    str = getcwd(path, sizeof(path));
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, buf) == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, buf);
+
+    /* change back to root unifyfs directory */
+    rc = chdir("..");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "..", strerror(errno));
+
+    /* check that we're in root unifyfs directory */
+    str = getcwd(path, sizeof(path));
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, unifyfs_root) == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, unifyfs_root);
+
+    /* change to unifyfs/dir1 directory using relative path */
+    rc = chdir("dir1");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "dir1", strerror(errno));
+
+    /* check that we're in unifyfs/dir1 directory */
+    str = getcwd(path, sizeof(path));
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, buf) == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, buf);
+
+    /* change to unifyfs/dir2 directory in strange way */
+    rc = chdir("././.././dir2");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "dir1", strerror(errno));
+
+    /* check that we're in unifyfs/dir2 directory */
+    str = getcwd(path, sizeof(path));
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, buf2) == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, buf2);
+
+
+#if 0
+    /* change to root directory */
+    rc = chdir("/");
```

Comment:
Need this commented out section?

Reply:
Ideally, it'd be nice to include these tests so long as the compiler doesn't complain about getwd.  I've added a TODO to explain why these are commented.  I'm hoping we might figure out how to conditionally allow them for the compilers that allow it.

Reply:
Ah, yeah. Didn't look at what tests they were, just saw they were commented out. Sounds good.